private/buf/bufworkspace/workspace_targeting.go

```diff
@@ -557,10 +562,28 @@ func getMappedModuleBucketAndModuleTargeting(
 			),
 		)
 	}
+	docBucket := bufmodule.GetDocStorageReadBucket(ctx, moduleBucket)
```

Comment:
This whole section needs comments, almost above every line - what is the exact logic here? Explain this code as if you were explaining it out loud to someone, but in the comments.